docs/evaluation/formal-math.md

```diff
@@ -1,6 +1,68 @@
 # Math (formal language)
 
-More details are coming soon!
+We support formal-math evaluation in Lean 4. The task is to generate a Lean 4 proof of a given formal statement.
+
+Under the hood we assemble the final Lean 4 program and evaluate it in a sandboxed Lean 4 environment, capturing a proof status for each item and writing it back to the originating files.
+
+## How we execute and evaluate
+
+Evaluation constructs a complete Lean 4 file from the model output and dataset metadata, then uses a sandboxed Lean 4 checker to validate it.
+
+At a high level, for each JSONL line in your result manifests we do the following:
+
+1. Build the final Lean program to check.
+   All the parameters mentioned here are *evaluation* parameters which are controlled with `--extra_eval_args="++eval_config.<parameter_name>=<parameter_value>`
+      - Normalize the model output to keep only the intended Lean code. By default we extract the last Lean code block when multiple are present (controlled by `extract_code_mode` parameter, which defaults to `"last"`). You can change the split phrase with `final_answer_key` parameter. If the `final_answer_key` phrase isn't present, we use the full generation instead.
```

Comment:
should we disable final_answer_key logic by default? I think it's not relevant to latest models